src/test/cursor.rs

```diff
@@ -0,0 +1,86 @@
+use std::time::Duration;
+
+use bson::doc;
+use futures::{future::Either, StreamExt};
+
+use crate::{
+    options::{CreateCollectionOptions, CursorType, FindOptions},
+    test::{TestClient, LOCK},
+    RUNTIME,
+};
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn tailable_cursor() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = TestClient::new().await;
+    client
+        .drop_collection(function_name!(), function_name!())
+        .await;
+
+    let db = client.database(function_name!());
+
+    db.create_collection(
+        function_name!(),
+        CreateCollectionOptions::builder()
+            .capped(true)
+            .max(5)
+            .size(1_000_000)
+            .build(),
+    )
+    .await
+    .unwrap();
+
+    let coll = db.collection(function_name!());
+
+    coll.insert_many((0..5).map(|i| doc! { "_id": i }), None)
+        .await
+        .unwrap();
+
+    let await_time = Duration::from_millis(500);
+
+    let mut cursor = coll
+        .find(
+            None,
+            FindOptions::builder()
+                .cursor_type(CursorType::TailableAwait)
+                .max_await_time(await_time)
```

Comment:
do we even want to allow users to customize this option anymore? It doesn't really have an effect on them anymore. Also, with client side operation timeout right around the corner, it could be soon deprecated anyways.

cc @mbroadst

Reply:
Do you mean specifically `TailableAwait`, or `CursorType` in general? The latter is still needed so the server knows not to close the cursor when the results are exhausted.

Reply:
I was referring to `max_await_time`.

Reply:
Ah, I misunderstood; sorry about that. I think it's reasonable to leave, since it determines the frequency of the `getMore` operations that the driver sends while waiting in a `Cursor::next` call, and some users might reasonably have preferences about that.

Reply:
Yeah I think for now it makes sense to leave it in, and only remove it if it gets deprecated as part of client side operation timeout.